src/persistence/profile.cpp

```diff
@@ -520,12 +525,19 @@ void Profile::setPassword(QString newPassword)
 {
     QList<HistoryKeeper::HistMessage> oldMessages = HistoryKeeper::exportMessagesDeleteFile();
     QByteArray avatar = loadAvatarData(core->getSelfId().publicKey);
-
+    QString oldPassword = password;
     password = newPassword;
     passkey = *core->createPasskey(password);
     saveToxSave();
-
     HistoryKeeper::getInstance()->importMessages(oldMessages);
     Nexus::getDesktopGUI()->reloadHistory();
     saveAvatar(avatar, core->getSelfId().publicKey);
+
+    QVector<uint32_t> friendList = core->getFriendList();
+    QVectorIterator<uint32_t> i(friendList);
+    while (i.hasNext())
+    {
+        QString friendPublicKey = core->getFriendPublicKey(i.next());
+        saveAvatar(loadAvatarData(friendPublicKey,oldPassword),friendPublicKey);
```

Comment:
If we previously had an empty password (unencrypted account), and we just set one, loadAvatarData will call avatarPath, which will see that we have a password set and return the path for an encrypted avatar, when we want to load the old unencrypted avatar here.

You could for example replace the "forceUnencrypted" parameter of avatarPath with an optional password parameter, and fix loadAvatarData to use it